### PR TITLE
[fix]: [CDS-49518]: NPE fix for Artifactory Repo Url

### PR DESCRIPTION
**Pull request by vtxorxwitty**
## Describe your changes
NPE issue resolved in case of Artifactory repoUrl mandatory change.  UT coverage to be picked up in this ticket https://harness.atlassian.net/browse/CDS-50899

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42478)

### CODE REVIEW

**Comment by vtxorxwitty** [comment]
trigger messagecheck

**Comment by vtxorxwitty** [comment]
trigger messagecheck

**Comment by vtxorxwitty** [comment]
trigger ss

File: 120-ng-manager/src/main/java/io/harness/ng/core/service/resources/ServiceResourceV2.java [hunk 1]
```diff
@@ -700,35 +697,92 @@ public ResponseDTO<ServiceInputsMergedResponseDto> mergeServiceInputs(
...
+
+    for (int i = 0; i < sourcesResMap.size(); i++) {
+      LinkedHashMap<String, Object> source = sourcesResMap.get(i);
+
+      String type = String.valueOf(source.get("type"));
+      type = type.substring(1, type.length() - 1);
+      LinkedHashMap<String, Object> spec = (LinkedHashMap<String, Object>) source.get("spec");
+
+      if (type.equals("ArtifactoryRegistry")) {
+        if (!spec.containsKey("repositoryUrl")) {
```
Comment (shivamnegi94) [inline 120-ng-manager/src/main/java/io/harness/ng/core/service/resources/ServiceResourceV2.java:755]: repositoryURL been used in multiple places, declare it as constant variable and use it

File: 120-ng-manager/src/main/java/io/harness/ng/core/service/resources/ServiceResourceV2.java [hunk 2]
```diff
@@ -700,35 +697,92 @@ public ResponseDTO<ServiceInputsMergedResponseDto> mergeServiceInputs(
...
+          String finalUrl = null;
+          String connectorRef = String.valueOf(spec.get("connectorRef"));
+          connectorRef = connectorRef.substring(1, connectorRef.length() - 1);
+          String repository = String.valueOf(spec.get("repository"));
+          repository = repository.substring(1, repository.length() - 1);
+          String repositoryFormat = String.valueOf(spec.get("repositoryFormat"));
+          repositoryFormat = repositoryFormat.substring(1, repositoryFormat.length() - 1);
+
+          if (repositoryFormat.equals("docker")) {
```
Comment (shivamnegi94) [inline 120-ng-manager/src/main/java/io/harness/ng/core/service/resources/ServiceResourceV2.java:764]: Please existing constant variable or enum for docker

**Review by sribalij — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42478)* status: 2 file(s) to review. 1 of 2 approvals obtained.

**Review by shivamnegi94 — APPROVED** [review]

**Review by shivamnegi94 — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42478)* status: :shipit: complete! ✓

**Review by archit-harness — APPROVED** [review]
> Reviewed 1 of 2 files at r2, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42478)* status: :shipit: complete! ✓

**Pull request merged**